git-secrets-scan-action/entrypoint.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+set -e
+
+# Clone and install git-secrets from source
+cd ${HOME}
+git clone https://github.com/awslabs/git-secrets.git && cd git-secrets && make install 
+
+cd ${GITHUB_WORKSPACE}
+
+# Register git-secrets in the workspace repo
+git secrets --install
+git secrets --register-aws
+
+set +e
+
+# Scan for secrets being committed
+if git secrets --scan 2>&1 >/dev/null | grep -q "[ERROR]"; then
```

Comment:
Let's do `--scan-history` as well.

Reply:
`--scan-history` actually takes a very long time, especially if the commit history is too large. I am not sure if that is required as part of the CI workflow.. we can do a manual `--scan-history` once before merging this change. All the changes following this should be checked by the CI

Reply:
Oh, interesting.

I think it is a legitimate concern though, because if we push multiple commits and some intermediate commit has credentials, then `git-secrets` won't to catch it but it will still be accessible by someone who's detail oriented.

Does `--scan-history` take longer than build and bundle? What if we made `git-secrets` run in parallel to our CI, instead of as a serial step in our CI?

Reply:
Oh yep, that is definitely a concern, good point. 

I tested `--scan-history` on our SAs(which is primarily our use case) and it doesn't take a lot of time(< 30s). I believe for our potential future use-cases, the time should be similar.  In that case, let us keep `--scan-history` in the action and run it sequentially in our workflow. In case we see issues with significantly larger CI times in the future we can find a workaround at that time.